### DISCUSSION TITLE
Can it be Locked By Signature with PyHanko?

### DISCUSSION THREAD

**Discussion by projackRK** [MatthiasValvekens/pyHanko#207] [Q&A] [answered]
I want to be able to sing more than once and be able to lock the next edit and sign as shown below. Can I use it in pyHanko?

**Comment by projackRK** [comment, marked as answer]
ok i found it for anyone wanting the same thing This might be a reply
Add
field_mdp_spec=fields.FieldMDPSpec(fields.FieldMDPAction.ALL),
doc_mdp_update_value=fields.MDPPerm.NO_CHANGES,
In Function "append_signature_field".

**Reply by MatthiasValvekens** [reply]
This should work, but be aware that it's not a particularly well-documented feature of the standard, so sometimes strange things happen.

Also, viewer support for locking in approval signatures (=> not certification signatures) is not very widespread. It's technically a PDF 2.0 feature.
